src/store/ducks/generic/MDAChildReport.ts

```diff
@@ -0,0 +1,167 @@
+import intersect from 'fast_array_intersect';
+import { AnyAction, Store } from 'redux';
+import { createSelector } from 'reselect';
+import SeamlessImmutable from 'seamless-immutable';
+
+/** the reducer name */
+export const reducerName = 'MDAChildReport';
+
+/** Child report interface */
+export interface ChildReport {
+  [key: string]: any;
+}
+
+// actions
+
+/** MDA_CHILD_REPORT_FETCHED action type */
+export const MDA_POINT_CHILD_REPORT_FETCHED =
+  'reveal/reducer/MDAPoint/MDAChildReport/MDA_Point_CHILD_REPORT_FETCHED';
+
+/** MDA_CHILD_REPORT_FETCHED action type */
+export const REMOVE_MDA_CHILD_REPORT_PLANS =
+  'reveal/reducer/MDAPoint/MDAChildReport/REMOVE_MDA_POINT_CHILD_REPORT';
+
+/** MDA_CHILD_REPORT_FETCHED action type */
+export const ADD_MDA_CHILD_REPORT_PLAN =
+  'reveal/reducer/MDAPoint/MDAChildReport/ADD_MDA_CHILD_REPORT';
+
+/** interface for fetch MDAPointChildReport action */
+interface FetchMDAPointChildReport extends AnyAction {
+  childReportByPlanId: { [key: string]: ChildReport[] };
+  type: typeof MDA_POINT_CHILD_REPORT_FETCHED;
+}
+
+/** interface for removing MDAPointChildReport action */
+interface RemoveMDAPointChildReportAction extends AnyAction {
+  childReportByPlanId: { [key: string]: ChildReport[] };
+  type: typeof REMOVE_MDA_CHILD_REPORT_PLANS;
+}
+
+/** Create type for MDAPointChildReportPlan reducer actions */
+export type MDAPointChildReportActionTypes =
+  | FetchMDAPointChildReport
+  | RemoveMDAPointChildReportAction
+  | AnyAction;
+
+/**
+ * Fetch Plan Definitions action creator
+ * @param {MDAPointChild[]} MDAPointPlanList - list of child report objects
+ */
+export const FetchMDAPointChildReportAction = (
+  childReportList: ChildReport[] = []
+): FetchMDAPointChildReport => {
+  const childReportByPlanId = {};
+  childReportList.forEach((report: ChildReport) => {
+    (childReportByPlanId as any)[report.plan_id]
+      ? (childReportByPlanId as any)[report.plan_id].push(report)
+      : ((childReportByPlanId as any)[report.plan_id] = [report]);
+  });
```

Comment:
lets refactor the use of any here, seems we already have a type for `childReportByPlanId`, ` { [key: string]: ChildReport[] }`